packages/api/src/middleware/validation/checkTask.js

```diff
@@ -0,0 +1,168 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import TaskModel from '../../models/taskModel.js';
+import { checkSoilAmendmentTaskProducts } from './checkSoilAmendmentTaskProducts.js';
+const adminRoles = [1, 2, 5];
+
+export function noReqBodyCheckYet() {
+  return async (req, res, next) => {
+    next();
+  };
+}
+
+const checkUpdateMiddlewareMap = {
```

Comment:
Love this file and doing these checks with middleware! ❤️ 

Just checking a few things regarding this map:

1. Didn't quite follow what you meant here:

> You can see I had to call the other middleware in the checkComplete() due to the fact that in complete task it is optional to provide task details, but in other endpoints it is mandatory.

But as written, isn't this equivalent to just calling the middleware in sequence on the route?

```js
router.patch(
  '/complete/soil_amendment_task/:task_id',
  modelMapping['soil_amendment_task'],
  hasFarmAccess({ params: 'task_id' }),
  checkScope(['edit:task']),
  checkCompleteTask(), // without using the map
  checkSoilAmendmentTaskProducts(),
  taskController.completeTask('soil_amendment_task'),
);
```

I don't mind the map (as a reminder for which task types don't have a body content check in place yet) but I don't understand the task details part of what you said.

2. Just curious when you were thinking this middleware would go on all the other task routes. Will it be a ticket/task for right after release?

Reply:
So I think as it is written if the `soil_amendment_task: {..}` was not provided in `task:{..}`. Then `checkSoilAmendmentTask()` would fail every time it is not provided. If it were chained on the route. Because that runs the check every time.

Since you can select "No changes were made to this task" that is a possibility.

This line bypasses `checkSoilAmendmentTask()` and moves to the controller if not defined:
```
 if (taskType in req.body) {
        checkUpdateMiddlewareMap[taskType]()(req, res, next);
      } else {
        next();
      }
```

That line was present in the previous middleware that was deleted `createOrPatchProduct()`: https://github.com/LiteFarmOrg/LiteFarm/pull/3208/commits/9de705c304b924c45aaa4b54aad2ca2a85e9cd9c

Reply:
Ohhhhh, okay. That makes sense, thank you 🙏

Reply:
Oh forgot to respond to 2. 

I have been wondering if we should create release epics on JIRA for future releases so if we see something that should be added we can make a ticket and evaluate it in there. 

I see this as part of tasks redesign personally. Or if we have the chance to do some tech debt and firm up our API.